### PR TITLE
ft: assigning bus to route 

### PR DESCRIPTION
**Pull request by christianR15c**
#### API for viewing all busses assigned to a certain route is: `/api/routes/buses/routeId` and replace the routeId by the route you want to view it's assigned busses

### CODE REVIEW

**Review by NyiingangoPacifique — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]

File: src/routes/assignBusToRoute.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import express from 'express'
+import Buses from '../controllers/assignBusToRoute'
+
+const router = express.Router()
+
+router.put('/api/routes/:routeId/:busId', Buses.assignBus)
+router.get('/api/routes/buses/:routeId', Buses.busAssociatedToRoute)
+
```
Comment (NyiingangoPacifique) [inline src/routes/assignBusToRoute.js]: you can change assign bus endpiont as suggested 
ex: /api/routes/:routeId/bus/:busId

**Review by NyiingangoPacifique — CHANGES_REQUESTED** [review]

File: src/routes/assignBusToRoute.js (hunk 1 above)
Comment (christianR15c) [reply src/routes/assignBusToRoute.js]: please review the changes you requested

**Review by NyiingangoPacifique — APPROVED** [review]

**Review by NyiingangoPacifique — APPROVED** [review]

**Review by Julesbi — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -37,3 +38,7 @@ get welcome page
 
 >>>>>>> setup project postgres
 >>>>>>> setup project postgres
+=======
```
Comment (SimonImanigirimpuhwe) [inline README.md]: Remove this lines

File: README.md [hunk 3]
```diff
@@ -37,3 +38,7 @@ get welcome page
...
+
+[![CI](https://github.com/atlp-rwanda/Phantom-Backend-Pirates/actions/workflows/main.yml/badge.svg?branch=develop)](https://github.com/atlp-rwanda/Phantom-Backend-Pirates/actions/workflows/main.yml)
+>>>>>>> e44ce18 (ft: assigning bus to route added)
```
Comment (SimonImanigirimpuhwe) [inline README.md]: And here

File: src/app.js [hunk 4]
```diff
@@ -1,107 +1,91 @@
...
+app.use(viewBus);
+app.use(userRoute);
+app.use(roleRouter);
+app.use(registerEmployeesRoute);
 
-// hostname
-const hostname = 'localhost'
+// port & hostname
+const port = process.env.APP_PORT || 3000;
+const hostname = 'localhost';
```
Comment (SimonImanigirimpuhwe) [inline src/app.js]: Please this should be environment variable

File: src/controllers/company.js [hunk 5]
```diff
@@ -45,28 +46,53 @@ class Companies {
 
   // list all Company
   static listAll (req, res) {
+<<<<<<< HEAD
```
Comment (SimonImanigirimpuhwe) [inline src/controllers/company.js]: Remove this line.

File: test/role.js [hunk 6]
```diff
@@ -32,15 +87,45 @@ describe('Testing role APIs', () => {
           res.body.should.be.a('object');
           res.body.should.have.property('id');
           res.body.should.have.property('role');
+<<<<<<< HEAD
```
Comment (SimonImanigirimpuhwe) [inline test/role.js]: And here

File: test/role.js [hunk 7]
```diff
@@ -53,24 +138,57 @@ describe('Testing role APIs', () => {
...
               response.body.should.be.a('object');
               response.body.should.have.property('message');
-              response.body.should.have.property('role');
               done();
             });
         });
     });
-  });
 
+<<<<<<< HEAD
```
Comment (SimonImanigirimpuhwe) [inline test/role.js]: Remove this line too

File: test/route.js [hunk 8]
```diff
@@ -32,8 +32,13 @@ describe('Routes API', () => {
 
   /* Test the GET by ID route */
   describe('GET /api/routes/:id', () => {
+<<<<<<< HEAD
     it('It should GET a route by ID', async() => {
       const routeId =1
+=======
+    it('It should GET a task by ID', (done) => {
+      const routeId = 28
+>>>>>>> 6ba7ade (ft: assigning bus to route added)
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Remove this line

File: test/route.js [hunk 9]
```diff
@@ -48,7 +53,11 @@ describe('Routes API', () => {
         })
     })
     it('It should NOT GET the route with wrong id', (done) => {
+<<<<<<< HEAD
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Here too

File: test/route.js [hunk 10]
```diff
@@ -48,7 +53,11 @@ describe('Routes API', () => {
...
         const routeId = 4
+=======
+        const routeId = 17
+>>>>>>> 6ba7ade (ft: assigning bus to route added)
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Here too

File: test/route.js [hunk 11]
```diff
@@ -121,10 +145,47 @@ describe('Routes API', () => {
...
+   describe('DELETE /api/routes/:id', () => {
+    it('It should DELETE a route', (done) => {
+      const routeId = 28
+      chai.request(API)
+        .delete('/api/routes/' + routeId)
+        .end((err, response) => {
+          response.should.have.status(200),
+          response.body.message.should.be.equal('route successfully deleted')
+
+>>>>>>> 6ba7ade (ft: assigning bus to route added)
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Remove this line

File: test/route.js [hunk 12]
```diff
@@ -121,10 +145,47 @@ describe('Routes API', () => {
...
           done()
         })
     })
   }) 
+<<<<<<< HEAD
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Please remove this line

File: test/route.js [hunk 13]
```diff
@@ -152,3 +213,6 @@ describe('Routes API', () => {
 });
 });
 });
+=======
+})
+>>>>>>> 6ba7ade (ft: assigning bus to route added)
```
Comment (SimonImanigirimpuhwe) [inline test/route.js]: Please remove this line as well

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Pull request merged**